packages/sitecore-jss/src/feaas/themes.test.ts

```diff
@@ -0,0 +1,365 @@
+import { expect } from 'chai';
+import { FEAAS_CDN, getFEAASLibraryStylesheetURLs, getStylesheetUrl } from './themes';
+
+describe('utils', () => {
+  describe('getFEAASLibraryStylesheetURLs', () => {
+    it('should return empty array route data is not provided', () => {
+      expect(
+        getFEAASLibraryStylesheetURLs({
+          sitecore: {
+            context: {},
+            route: null,
+          },
+        })
+      ).to.deep.equal([]);
+    });
+
+    describe('normal mode', () => {
+      it('should return urls using CSSStyles field', () => {
+        expect(
+          getFEAASLibraryStylesheetURLs({
+            sitecore: {
+              context: {},
+              route: {
+                name: 'home',
+                fields: {
+                  CSSStyles: {
+                    value: '-library--foo',
+                  },
+                  LibraryId: {
+                    value: 'bar',
+                  },
+                },
+                placeholders: {},
+              },
+            },
+          })
+        ).to.deep.equal([getStylesheetUrl('foo')]);
+      });
+
+      it('should return urls using LibraryId field', () => {
+        expect(
+          getFEAASLibraryStylesheetURLs({
+            sitecore: {
+              context: {},
+              route: {
+                name: 'home',
+                fields: {
+                  LibraryId: {
+                    value: 'bar',
+                  },
+                },
+                placeholders: {},
+              },
+            },
+          })
+        ).to.deep.equal([getStylesheetUrl('bar')]);
+      });
+
+      it('should return urls using custom hostname', () => {
+        expect(
+          getFEAASLibraryStylesheetURLs(
+            {
+              sitecore: {
+                context: {},
+                route: {
+                  name: 'home',
+                  fields: {
+                    LibraryId: {
+                      value: 'bar',
+                    },
+                  },
+                  placeholders: {},
+                },
+              },
+            },
+            'https://foo.net'
+          )
+        ).to.deep.equal([getStylesheetUrl('bar', 'https://foo.net')]);
+      });
+
+      it('should return empty urls array when required fields are not provided', () => {
+        expect(
+          getFEAASLibraryStylesheetURLs({
+            sitecore: {
+              context: {},
+              route: {
+                name: 'home',
+                fields: {},
+                placeholders: {},
+              },
+            },
+          })
+        ).to.deep.equal([]);
+      });
+
```

Comment:
Looks it's covered in the "should traverse nested nodes and return only unique urls" test, but should we add a dedicated test for "should prefer library in CSSStyles field over LibraryId field" to cover [this](https://github.com/Sitecore/jss/pull/1359/files#diff-311766df4ca59b54ee037541af1783f000ce1e943dc27859e7a12fb8617c2d98R70-R74)?

Reply:
@ambrauer Such case is covered [here](https://github.com/Sitecore/jss/pull/1359/files#diff-e47f735be035cfc699f6a896b739f57ad82f99a7320eaea1d355ccc6586adc25R26-R31)

Reply:
And this test is covered, but I decided to have a separate test to highlight this case